src/server/cmd/pachd/main.go

```diff
@@ -711,25 +712,25 @@ func doFullMode(ctx context.Context, config interface{}) (retErr error) {
 	}))
 	eg.Go(maybeIgnoreErrorFunc("S3 Server", requireNoncriticalServers, func() error { return s3Server{env.GetPachClient, env.Config().S3GatewayPort}.listenAndServe(ctx) }))
 	eg.Go(maybeIgnoreErrorFunc("Prometheus Server", requireNoncriticalServers, func() error { return prometheusServer{port: env.Config().PrometheusPort}.listenAndServe(ctx) }))
-	go func(c chan os.Signal) {
-		<-c
-		log.Println("terminating; waiting for pachd server to gracefully stop")
-		var g, _ = errgroup.WithContext(ctx)
-		g.Go(func() error { externalServer.Server.GracefulStop(); return nil })
-		g.Go(func() error { internalServer.Server.GracefulStop(); return nil })
-		if err := g.Wait(); err != nil {
-			log.Errorf("error waiting for pachd server to gracefully stop: %v", err)
+	eg.Go(func() error {
```

Comment:
should this logic be in the other modes?

Reply:
It was originally only in full & paused, not in enterprise or sidecar. I figured that I would leave things as they are, particularly because they’ll get unified in CORE-825.